src/memory/AmdFlash.cc

```diff
@@ -652,18 +696,82 @@ bool AmdFlash::checkCommandEraseChip()
 
 			for (const Sector& sector : sectors) {
 				if (isWritable(sector)) {
-					ram->memset(sector.writeAddress, 0xff, sector.size);
+					erase.markPending(sector);
 				} else {
 					statusRegister.sectorLock = true;
 				}
 			}
 
-			status.dataPolling = true; // immediate completion
+			statusRegister.ready = false;
+			status.dataPolling = false;
+			setState(State::ERASE_CHIP);
+			scheduleEraseOperation(time);
 		}
 	}
 	return false;
 }
 
+void AmdFlash::scheduleEraseOperation(EmuTime::param time)
+{
+	if (state == State::ERASE_SECTOR && !status.eraseTimer) {
+		syncOperation.removeSyncPoint();
+		syncOperation.setSyncPoint(time + chip.erase.timerDuration);
+	} else if (std::ranges::any_of(erase.buffer, std::identity{})) {
```

Comment:
If my assumption below is correct (that logically this is a `vector<bool>`), you could write this using our `contains()` helper function in `stl.hh`.
```
contains(erase.buffer, true)
```
I'm not sure which is more clear. Pick whichever you like best.